tests/testing/strategies_test.py

```diff
@@ -0,0 +1,182 @@
+    assert var.ndim == 3
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(sizes={}))
+def test_variables_can_set_sizes_empty(var):
+    assert var.sizes == {}
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(sizes={'abc': 2}))
+def test_variables_can_set_sizes_1d(var):
+    assert var.sizes == {'abc': 2}
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(sizes=st.just({'def': 3})))
+def test_variables_can_set_sizes_strategy(var):
+    assert var.sizes == {'def': 3}
+
+
+def test_variables_ndim_and_sizes_are_mutually_exclusive():
+    @settings(max_examples=1, database=None)
+    @given(scst.variables(ndim=1, sizes={'a': 2}))
+    def draw_from_strategy(_):
+        pass
+
+    with pytest.raises(InvalidArgument):
+        draw_from_strategy()
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(unit='one'))
+def test_variables_can_set_unit_to_one(var):
+    assert var.unit == 'one'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(unit='m'))
+def test_variables_can_set_unit_to_m(var):
+    assert var.unit == 'm'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(unit=st.just('s')))
+def test_variables_can_set_unit_strategy(var):
+    assert var.unit == 's'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(dtype='int64'))
+def test_variables_can_set_dtype_to_int64(var):
+    assert var.dtype == 'int64'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(dtype=float))
+def test_variables_can_set_dtype_to_float(var):
+    assert var.dtype == 'float64'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(dtype=st.just('int64')))
+def test_variables_can_set_dtype_strategy(var):
+    assert var.dtype == 'int64'
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(with_variances=True, dtype=scst.floating_dtypes()))
+def test_variables_can_toggle_variances_on(var):
+    assert var.variances is not None
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(with_variances=False, dtype=scst.floating_dtypes()))
+def test_variables_can_toggle_variances_off(var):
+    assert var.variances is None
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.variables(with_variances=st.just(True), dtype=scst.floating_dtypes()))
+def test_variables_can_set_variances_strategy(var):
+    assert var.variances is not None
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(scst.n_variables(2))
+def test_n_variables_all_have_same_parameters(variables):
+    a, b = variables
+    assert a.sizes == b.sizes
+    assert a.unit == b.unit
+    assert a.dtype == b.dtype
+    assert (a.variances is None) == (b.variances is None)
+
+
+@settings(max_examples=N_EXAMPLES)
+@given(
+    scst.dataarrays(data_args={
+        'dtype': int,
+        'sizes': {
+            'a': 2,
+            'b': 3
+        }
+    },
+                    coords=['b'],
+                    coord_args=None))
+def test_dataarrays(da):
+    print(da)
```

Comment:
Intentional?

Reply:
No. Apparently I forgot to implement this.
That reminds me, the `dataarrays` strategy is little more than an experimental prototype at this point. But I'd rather flesh it out in a follow up PR to not make this one more complicated.